torch/_subclasses/meta_utils.py

```diff
@@ -458,6 +461,10 @@ class MetaTensorDesc:
     is_gradtrackingtensor: bool = False
     is_view: bool = False
     is_nested: bool = False
+    # We eagerly symbolicize the associated nested int for e.g. offsets / lengths
+    # metadata if that offsets is already associated with a nested int.
+    # See test_construct_from_jagged_with_input_offsets_mixed_case.
+    nested_int: Optional[int] = None
```

Comment:
hmm... `Optional[int]` is wrong, right? (since this is a nested_int). But `nested_int` is a type of `SymInt`, and... putting `SymInt` here feels wrong since `MetaTensorDesc` is serializable but SymInts in general are not serializable.

Do we have another python type to tell nested ints apart from other SymInts? Or maybe SymInt is fine here just for type annotations

Reply:
I think the idea here is that we imagine nested int to be some POD uniquely identified by its "id", so we don't necessarily need to store a particular instance.

Reply:
(You were right in that we were actually saving the nested int object today, I've just updated to actually use an int)